pennylane/templates/subroutines/single_excitation_op.py

```diff
@@ -0,0 +1,176 @@
+    creation operators and the indices :math:`r` and :math:`p` run over the occupied and
+    unoccupied molecular orbitals, respectively. Using the `Jordan-Wigner transformation
+    <https://arxiv.org/abs/1208.5986>`_ the fermionic operator defined above can be written
+    in terms of Pauli matrices (for more details see
+    `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_).
+
+    .. math::
+
+        \hat{U}_{pr}^{(1)}(\theta) = \mathrm{exp} \Big\{ \frac{i\theta}{2}
+        \bigotimes_{a=r+1}^{p-1}\hat{Z}_a (\hat{Y}_r \hat{X}_p) \Big\}
+        \mathrm{exp} \Big\{ -\frac{i\theta}{2}
+        \bigotimes_{a=r+1}^{p-1} \hat{Z}_a (\hat{X}_r \hat{Y}_p) \Big\}.
+
+    The quantum circuit to exponentiate the tensor product of Pauli matrices entering
+    the latter equation is shown below:
+
+    .. figure:: ../../_static/ucc_se_op.png
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
+
+    As explained in `Seely et al. (2012) <https://arxiv.org/abs/1208.5986>`_,
+    the exponential of a tensor product of Pauli-Z operators can be decomposed in terms of
+    :math:`2(n-1)` CNOT gates and a single-qubit Z-rotation. If there are :math:`X` or
+    :math:`Y` Pauli matrices in the product, the Hadamard (:math:`H`) or :math:`R_x` gate has
+    to be applied to change to the :math:`X` or :math:`Y` basis, respectively.
+
+    Args:
+        weight (float): angle :math:`\theta` entering the Z rotation acting on wire ``p``
+        wires (sequence[int]): two-element sequence with the qubit indices ``r, p``
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        Notice that:
+
+        #. :math:`\hat{U}_{pr}^{(1)}(\theta)` involves two exponentiations where :math:`\hat{U}_1`,
+           :math:`\hat{U}_2`, and :math:`\hat{U}_\theta` are defined as follows,
+
+           .. math::
+               [U_1, U_2, U_{\theta})] = \Bigg\{\bigg[R_x(-\pi/2), H, R_z(\theta/2)\bigg],
+               \bigg[H, R_x(-\frac{\pi}{2}), R_z(-\theta/2) \bigg] \Bigg\}
+
+        #. For a given pair ``[r, p]``, ten single-qubit operations are applied. Notice also that
+           CNOT gates act only on qubits with indices between ``r`` and ``p``. The operations
+           performed across these qubits are shown in dashed lines in the figure above.
+
+    """
+
+    ##############
+    # Input checks
+
+    check_no_variable(wires, msg="'wires' cannot be differentiable")
+
+    wires = check_wires(wires)
+
+    expected_shape = (2,)
+    check_shape(
+        wires,
+        expected_shape,
+        msg="'wires' must be of shape {}; got {}" "".format(expected_shape, get_shape(wires)),
+    )
+
+    expected_shape = ()
+    check_shape(
+        weight,
+        expected_shape,
+        msg="'weight' must be of shape {}; got {}" "".format(expected_shape, get_shape(weight)),
+    )
+
+    check_type(wires, [list], msg="'wires' must be a list; got {}" "".format(wires))
```

Comment:
```suggestion
    check_type(wires, [list], msg="'wires' must be a list; got {}".format(wires))
```

Reply:
Actually, I would delete this line checking the wire type altogether - wires are already check in the special ``check_wires`` function, and users frequently feed range, tuple or lists (i.e. any iterable) to wires, which is no problem if you remove this check.